src/c_kzg_4844.c

```diff
@@ -1796,29 +1725,22 @@ C_KZG_RET load_trusted_setup(
         blst_p2_from_affine(&out->g2_values[i], &g2_affine);
     }
 
-    /* It's the smallest power of 2 >= n1 */
-    unsigned int max_scale = 0;
-    while ((1ULL << max_scale) < n1)
-        max_scale++;
-
     /* Initialize the KZGSettings struct */
-    ret = c_kzg_malloc((void **)&out->fs, sizeof(FFTSettings));
-    if (ret != C_KZG_OK) goto out_error;
-    ret = new_fft_settings(out->fs, max_scale);
-    if (ret != C_KZG_OK) goto out_error;
-    ret = fft_g1(out->g1_values, g1_projective, true, n1, out->fs);
+    ret = new_roots_of_unity(out->roots_of_unity, max_scale);
```

Comment:
IMO we should rename `new_roots_of_unity()` to `get_roots_of_unity()` or `compute_roots_of_unity()`. That's because `new_*()` functions are usually constructor functions and this one doesn't really construct something new.

Reply:
Good idea. Went with the `compute` one.